src/functTest/groovy/com/bmuschko/gradle/docker/DockerJavaApplicationPluginFunctionalTest.groovy

```diff
@@ -134,10 +207,10 @@ EXPOSE 9090 8080
 """FROM $CUSTOM_BASE_IMAGE
 MAINTAINER Benjamin Muschko "benjamin.muschko@gmail.com"
 ADD ${projectName}-1.0.tar /
-ENTRYPOINT ["/${projectName}-1.0/bin/${projectName}"]
-EXPOSE 9090
 ADD file1.txt /some/dir/file1.txt
 ADD file2.txt /other/dir/file2.txt
+ENTRYPOINT ["/${projectName}-1.0/bin/${projectName}"]
+EXPOSE 9090
```

Comment:
Most of this feedback overlaps in a way that leads me to rethink the design, I'll switch over to using a new Instruction class which can cross off several of these boxes.